files/build_templates/qos_config.j2

```diff
@@ -0,0 +1,209 @@
+            "7": "7"
+        }
+    },
+    "TC_TO_QUEUE_MAP": {
+        "AZURE": {
+            "0": "0",
+            "1": "1",
+            "2": "2",
+            "3": "3",
+            "4": "4",
+            "5": "5",
+            "6": "6",
+            "7": "7"
+        }
+    },
+    "DSCP_TO_TC_MAP": {
+        "AZURE": {
+            "0" : "0",
+            "1" : "0",
+            "2" : "0",
+            "3" : "3",
+            "4" : "4",
+            "5" : "0",
+            "6" : "0",
+            "7" : "0",
+            "8" : "1",
+            "9" : "0",
+            "10": "0",
+            "11": "0",
+            "12": "0",
+            "13": "0",
+            "14": "0",
+            "15": "0",
+            "16": "0",
+            "17": "0",
+            "18": "0",
+            "19": "0",
+            "20": "0",
+            "21": "0",
+            "22": "0",
+            "23": "0",
+            "24": "0",
+            "25": "0",
+            "26": "0",
+            "27": "0",
+            "28": "0",
+            "29": "0",
+            "30": "0",
+            "31": "0",
+            "32": "0",
+            "33": "0",
+            "34": "0",
+            "35": "0",
+            "36": "0",
+            "37": "0",
+            "38": "0",
+            "39": "0",
+            "40": "0",
+            "41": "0",
+            "42": "0",
+            "43": "0",
+            "44": "0",
+            "45": "0",
+            "46": "0",
+            "47": "0",
+            "48": "0",
+            "49": "0",
+            "50": "0",
+            "51": "0",
+            "52": "0",
+            "53": "0",
+            "54": "0",
+            "55": "0",
+            "56": "0",
+            "57": "0",
+            "58": "0",
+            "59": "0",
+            "60": "0",
+            "61": "0",
+            "62": "0",
+            "63": "0"
+        }
+    },
+    "SCHEDULER": {
+        "scheduler.0": {
+            "type"  : "DWRR",
+            "weight": "25"
+        },
+        "scheduler.1": {
+            "type"  : "DWRR",
+            "weight": "30"
+        },
+        "scheduler.2": {
+            "type"  : "DWRR",
+            "weight": "20"
+        }
+    },
+{% if pfc_to_pg_map is defined and pfc_to_pg_map %}
+    "PFC_PRIORITY_TO_PRIORITY_GROUP_MAP": {
+        "AZURE": {
+            "0": "0",
+            "1": "1",
+            "3": "3",
+            "4": "4"
+        }
+    },
+{% endif %}
+    "PORT_QOS_MAP": {
+        "{{ port_names_active }}": {
+            "dscp_to_tc_map"  : "[DSCP_TO_TC_MAP|AZURE]",
+            "tc_to_queue_map" : "[TC_TO_QUEUE_MAP|AZURE]",
+            "tc_to_pg_map"    : "[TC_TO_PRIORITY_GROUP_MAP|AZURE]",
+            "pfc_to_queue_map": "[MAP_PFC_PRIORITY_TO_QUEUE|AZURE]",
+{% if pfc_to_pg_map is defined and pfc_to_pg_map %}
+            "pfc_to_pg_map"   : "[PFC_PRIORITY_TO_PRIORITY_GROUP_MAP|AZURE]",
+{% endif %}
+            "pfc_enable"      : "3,4"
+        }
+    },
+    "WRED_PROFILE": {
```

Comment:
Why it does not apply? Let me try on msn2700

Reply:
It works on msn2700. Other than a slight drift of the min threshold (1084 -> 1088) and max threshold (3250 -> 3264) value after dividing the cell size, I do not see any problem. Can you elaborate? Also, where does the drift come from?


***********************************************
 RED/ECN
***********************************************

General Configurations
..............................
Module initialized                       TRUE
deafult_red_enabled                      TRUE
free profiles                            3066
used profiles                            6
default_scd                              FALSE
default_aqs_weight                       0
default_aqs_time                         0
default_count_marked                     FALSE
default_count_marked                     FALSE
default_red_enabled                      TRUE
default_tc_ecn_enabled                   FALSE
default_tc_wred_enabled                  FALSE

Used Profiles
..............................

--------------------------------------------------------------------------------------
    Id| #mappings|                     Mode|  High_drop_percent|       Min|       Max|
--------------------------------------------------------------------------------------
     0|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------
     1|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------
     2|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------
     3|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------
     4|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------
     5|       128|     REDECN_MODE_ABSOLUTE|                100|      1088|      3264|
--------------------------------------------------------------------------------------

Egress Port Configurations
..............................

Port 65792 Data
..............................
Port 65792 Tc Enabled Table
..............................

--------------------------------------------
  Tc|     Ecn|     Red|                Mode|
--------------------------------------------
   0|    TRUE|    TRUE|REDECN_MODE_ABSOLUTE|
--------------------------------------------
   1|    TRUE|    TRUE|REDECN_MODE_ABSOLUTE|
--------------------------------------------
   3|    TRUE|    TRUE|REDECN_MODE_ABSOLUTE|
--------------------------------------------
   4|    TRUE|    TRUE|REDECN_MODE_ABSOLUTE|
--------------------------------------------

Port 65792 Bindings
..............................

--------------------------------------------------------------------------------------------------------------------------------
   Port_id|   Tc|                Flow|  Hw_profile|  Profile_id|                     Mode|  High_drop_percent|     Min|     Max|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|           TCP_GREEN|           1|           3|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|          TCP_YELLOW|           2|           4|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|             TCP_RED|           3|           5|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|       NON_TCP_GREEN|           1|           3|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|      NON_TCP_YELLOW|           2|           4|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    0|         NON_TCP_RED|           3|           5|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|           TCP_GREEN|           1|           3|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|          TCP_YELLOW|           2|           4|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|             TCP_RED|           3|           5|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|       NON_TCP_GREEN|           1|           3|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|      NON_TCP_YELLOW|           2|           4|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    1|         NON_TCP_RED|           3|           5|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|           TCP_GREEN|           1|           0|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|          TCP_YELLOW|           2|           1|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|             TCP_RED|           3|           2|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|       NON_TCP_GREEN|           1|           0|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|      NON_TCP_YELLOW|           2|           1|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    3|         NON_TCP_RED|           3|           2|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|           TCP_GREEN|           1|           0|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|          TCP_YELLOW|           2|           1|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|             TCP_RED|           3|           2|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|       NON_TCP_GREEN|           1|           0|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|      NON_TCP_YELLOW|           2|           1|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
0x10100   |    4|         NON_TCP_RED|           3|           2|     REDECN_MODE_ABSOLUTE|                100|    1088|    3264|
--------------------------------------------------------------------------------------------------------------------------------
port rate_based configured               FALSE

Reply:
on Mellanox switches min/max should be multiply of 6144 (64*96). If it is not - it will be rounded. Because of this sometimes (when min is close or equal to max) values cannot be re-applied... 

About unifying - if buffers are configured for different size on different platforms does it make sense to have the same ECN settings for all?

Reply:
Thx, it makes sense now. I think the ECN settings are independent to MMU settings. We will make changes to mlnx mmu settings soon, as required by the external team. But, we should see it as a coincidence.